File: src/main/java/com/back/nbe9112team06/domain/timetable/entity/DateInfo.java
package com.back.nbe9112team06.domain.timetable.entity;

import com.back.nbe9112team06.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
public class DateInfo extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "time_table_id")
    private TimeTable timeTable;

    private LocalDate date;

    @OneToMany(mappedBy = "dateInfo", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<TimeInfo> timeInfos = new ArrayList<>();


    public DateInfo(TimeTable timeTable, LocalDate date) {
        this.timeTable = timeTable;
        this.date = date;
    }
}

File: src/main/kotlin/com/back/nbe9112team06/domain/timetable/entity/DateInfo.kt
package com.back.nbe9112team06.domain.timetable.entity

import com.back.nbe9112team06.global.entity.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.BatchSize
import java.time.LocalDate

@Entity
class DateInfo(
    @JoinColumn(name = "time_table_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var timeTable: TimeTable,
    var date: LocalDate
) : BaseEntity() {

    @OneToMany(mappedBy = "dateInfo", cascade = [CascadeType.ALL], orphanRemoval = true)
    @BatchSize(size = 100)
    val timeInfos: MutableList<TimeInfo> = mutableListOf()
}
